src/main/java/utils/StringToDate.java

```diff
@@ -0,0 +1,141 @@
+package utils;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * This class returns literal dates converted to date format.
+ */
+public class StringToDate {
+
+    SimpleDateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+
+    /**
+     * Converts literal date String to a date format.
+     *
+     * @param stringToConvert a literal date String.
+     * @return a date format.
+     */
+    public Date convert(final String stringToConvert) {
+        if (stringToConvert == null) {
+            throw new IllegalArgumentException("Value is null");
+        }
+        if (stringToConvert.equals("")) {
+            throw new IllegalArgumentException("Value is empty");
+        }
+
+        if (stringToConvert.matches("TODAY|TOMORROW|YESTERDAY")) {
+            return dayName(stringToConvert);
+        }
+
+        if (stringToConvert.contains("ago") || stringToConvert.contains("from now")) {
+            return getDateValue(stringToConvert);
+        }
+
+        if (stringToConvert.contains("/")) {
+            try {
+                return dateFormat.parse(stringToConvert);
+            } catch (ParseException exception) {
+                throw new IllegalArgumentException(exception);
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Converts 'TODAY, TOMORROW, YESTERDAY' to a date format.
+     *
+     * @param dateString a literal date String.
+     * @return a date format.
+     */
+    public Date dayName(final String dateString) {
+        if ("TODAY".equals(dateString)) {
+            return getTodayDate();
+        }
+        if ("TOMORROW".equals(dateString)) {
+            return getTimelineDate(1, Calendar.DATE);
+        }
+        if ("YESTERDAY".equals(dateString)) {
+            return getTimelineDate(- 1, Calendar.DATE);
+        }
+        return null;
```

Comment:
maybe its better to return an exception instead of null, or how are you managing the null values?